metricbeat/module/redis/info/info.go

```diff
@@ -171,20 +177,34 @@ func (m *MetricSeter) Setup(cfg *ucfg.Config) error {
 
 func (m *MetricSeter) Fetch(ms *helper.MetricSet) (events []common.MapStr, err error) {
 	for _, host := range ms.Config.Hosts {
-		c := m.redisPools[host].Get()
-		out, err := rd.String(c.Do("INFO"))
-		c.Close()
-		if err != nil {
-			logp.Err("Error converting to string: %v", err)
-		}
+		// Fetch default INFO
+		info := m.fetchRedisStats(host, "default")
+
+		// TODO: Take this approach for all sub types, or grep out keyspace from info
+		//
+		// Fetch keyspace metrics, so they can be easily iterated over later.
+		// It might be nice to take this approach for all sub stats from
+		// INFO and allow user to configure which sub metric sets they collect.
+		keyspace := m.fetchRedisStats(host, "keyspace")
```

Comment:
So you're suggesting to just make the one call, `info := m.fetchRedisStats(host, "default")`, and parse out the `keyspace` stats from there? 

My idea here, was too extend this to pull all metrics with the specific stat name, which would allow the users to configure each stat they want.